inference-engine/src/mkldnn_plugin/nodes/common/tensor_desc_creator.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ie_layouts.h>
+
+namespace MKLDNNPlugin {
+
+enum class TensorDescCreatorTypes : unsigned {
+    nspc,       // general per channels format
+    ncsp,        // general planar
+    nCsp8c,     // general channels blocked by 8
+    nCsp16c    // general channels blocked by 16
+};
+
+class CreatorsMapFilterConstIterator;
+
+class TensorDescCreator {
```

Comment:
Two things about TensorDescCreator and related logic:
1) We definitely lack brief description of its API.
2) It should be covered by unit tests (w/o connection to any graph patterns).

I think both tasks can be out of scope of this PR, but we should create follow up ticket.